pkg/tfvars/internal/cache/cache.go

```diff
@@ -232,5 +231,18 @@ func DownloadFile(baseURL string, dataType string) (string, error) {
 func DownloadImageFile(baseURL string) (string, error) {
 	logrus.Infof("Obtaining RHCOS image file from '%v'", baseURL)
 
-	return DownloadFile(baseURL, imageDataType)
+	var u urlWithIntegrity
+	parsedURL, err := url.ParseRequestURI(baseURL)
+	if err != nil {
+		return "", err
+	}
+	q := parsedURL.Query()
+	if uncompressedSHA256, ok := q["sha256"]; ok {
+		u.uncompressedSHA256 = uncompressedSHA256[0]
+		q.Del("sha256")
+		parsedURL.RawQuery = q.Encode()
```

Comment:
I don't know if this is a bug in go, but this is not preserving an encoded space.
https://play.golang.org/p/Teo3kCdijPi

Reply:
Nevermind. This is my poor understanding of URL encoding for query fragments. Apparently '+' is equivalent to space for a query value. And if you actually want '+' you need to use %2B.